bitutils/bitutils_test.go

```diff
@@ -0,0 +1,88 @@
+package bitutils
+
+import (
+	"testing"
+)
+
+// Define some test data
+
+// 0000 0100 1010 0010 0000 0011 1011 0001 0000 0000 0010 1011
```

Comment:
I think it's confusing to have the test data shared among all cases. Consider crafting example bits per case designed to easily detect errors at the index boundaries.

Reply:
I figured this setup would more resemble the "real world" case and better allow for deeper test examples (such as starting several bytes in to the array).

Reply:
Also writing out the bit field once at the start would save writing it out for each test, or having someone do the conversion in their head from hex in order to follow along.